src/Compilers/Core/Portable/SourceGeneration/GeneratorContexts.cs

```diff
@@ -70,8 +70,16 @@ internal GeneratorExecutionContext(Compilation compilation, ParseOptions parseOp
         /// Adds source code in the form of a <see cref="string"/> to the compilation.
         /// </summary>
         /// <param name="hintName">An identifier that can be used to reference this source text, must be unique within this generator</param>
-        /// <param name="source">The source code to be add to the compilation</param>
-        public void AddSource(string hintName, string source) => AddSource(hintName, SourceText.From(source, Encoding.UTF8));
+        /// <param name="source">The source code to add to the compilation</param>
+        public void AddSource(string hintName, string source) => AddSource(hintName, SourceText.From(source, Encoding.UTF8, SourceHashAlgorithm.Sha256));
```

Comment:
>SourceHashAlgorithm.Sha256 [](start = 124, length = 26)

I'd like to think through the implications of changing this slightly more before we do so. I agree we should use 256 below where we're forced to supply an algorithm, but I don't think we should make the change here as part of this PR.